pyproject.toml

```diff
@@ -13,35 +13,35 @@ readme = "README.md"
 license = { file = "LICENSE" }
 dependencies = [
     "bokeh==3.0.1",
-    "certifi==2022.9.24",
+    "certifi==2023.7.22",
     "charset-normalizer==2.1.1",
     "contourpy==1.0.6",
     "cycler==0.11.0",
-    "fonttools==4.38.0",
-    "idna==3.4",
-    "Jinja2==3.1.2",
+    "fonttools==4.43.0",
+    "idna==3.7",
+    "jinja2==3.1.4",
     "joblib==1.2.0",
     "kiwisolver==1.4.4",
     "MarkupSafe==2.1.1",
     "matplotlib==3.6.2",
     "numpy==1.23.4",
     "packaging==21.3",
     "pandas==1.5.1",
-    "Pillow==9.3.0",
+    "Pillow==10.3.0",
     "pyparsing==3.0.9",
     "PyQt5==5.15.10",
     "python-dateutil==2.8.2",
     "pytz==2022.6",
     "PyYAML==6.0",
-    "requests==2.28.1",
-    "scikit-learn==1.1.3",
+    "requests==2.32.0",
+    "scikit-learn==1.5.0",
```

Comment:
Are we using scikit-learn?  I think I removed it and implemented kmeans from scratch.  Maybe not needed.